java/src/jmri/jmrit/logixng/actions/ActionClockRate.java

```diff
@@ -0,0 +1,235 @@
+import java.beans.PropertyChangeListener;
+import java.util.*;
+
+import jmri.*;
+import jmri.jmrit.logixng.*;
+import jmri.jmrit.logixng.util.LogixNG_SelectEnum;
+import jmri.jmrit.logixng.util.LogixNG_SelectDouble;
+import jmri.jmrit.logixng.util.parser.ParserException;
+import static jmri.jmrit.simpleclock.SimpleTimebase.MAXIMUM_RATE;
+import static jmri.jmrit.simpleclock.SimpleTimebase.MINIMUM_RATE;
+import jmri.util.ThreadingUtil;
+
+/**
+ * This action provides the ability to set the fast clock speed.
+ *
+ * @author Daniel Bergqvist Copyright 2021
+ * @author Dave Sand Copyright 2021
+ * @author Daniel Bergqvist Copyright 2022
+ */
+public class ActionClockRate extends AbstractDigitalAction
+        implements PropertyChangeListener {
+
+    private final LogixNG_SelectEnum<ClockState> _selectEnum =
+            new LogixNG_SelectEnum<>(this, ClockState.values(), ClockState.SetClockSpeed, this);
+    private final LogixNG_SelectDouble _selectSpeed =
+            new LogixNG_SelectDouble(this, 3, this, new DefaultFormatterParserValidator());
+
+
+    public ActionClockRate(String sys, String user)
+            throws BadUserNameException, BadSystemNameException {
+        super(sys, user);
+    }
+
+    @Override
+    public Base getDeepCopy(Map<String, String> systemNames, Map<String, String> userNames) throws ParserException {
+        DigitalActionManager manager = InstanceManager.getDefault(DigitalActionManager.class);
+        String sysName = systemNames.get(getSystemName());
+        String userName = userNames.get(getSystemName());
+        if (sysName == null) sysName = manager.getAutoSystemName();
+        ActionClockRate copy = new ActionClockRate(sysName, userName);
+        copy.setComment(getComment());
+        _selectEnum.copy(copy._selectEnum);
+        _selectSpeed.copy(copy._selectSpeed);
+        return manager.registerAction(copy);
+    }
+
+    public LogixNG_SelectEnum<ClockState> getSelectEnum() {
+        return _selectEnum;
+    }
+
+    public LogixNG_SelectDouble getSelectSpeed() {
+        return _selectSpeed;
+    }
+
+    /**
+     * Convert speed to a decimal string.
+     * @param speed The speed
+     * @return speed formatted as %1.3f
+     */
+    public static String formatSpeed(double speed) {
+        return String.format("%1.3f", speed);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return Category.ITEM;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void execute() throws JmriException {
+
+        ClockState theState = _selectEnum.evaluateEnum(getConditionalNG());
+        double theValue = _selectSpeed.evaluateValue(getConditionalNG());
+
+        jmri.Timebase timebase = InstanceManager.getDefault(jmri.Timebase.class);
+
+        ThreadingUtil.runOnLayoutWithJmriException(() -> {
+            switch(theState) {
+                case SetClockSpeed:
+                    try {
+                        timebase.setRate(theValue);
```

Comment:
This should probably be `userSetRate`.  This avoids the error message and also updates IMRATEFACTOR.

Reply:
Thank you!